### PR TITLE
Adds PrintJson to bluemix/terminal/table

### PR DESCRIPTION
**Pull request by allmightyspiff**
I use this table library a lot in the `ibmcloud sl` plugin, and it would be nice to have a way to simply print the table as a JSON string. Currently my only option is to print the raw data that went into the table originally, which might not match up exactly with what I want to show the user.

My approach here was to just render the table as an array, each row being an object with each property matching a column in the table.

For example
```go
	testTable := NewTable(&buf, []string{"test1", "test2"})
	testTable.Add("row1-col1", "row1-col2")
	testTable.Add("row2-col1", "row2-col2")
	testTable.PrintJson()
```
would render something like this:

```json
[
        {
                "test1": "row1-col1",
                "test2": "row1-col2"
        },
        {
                "test1": "row2-col1",
                "test2": "row2-col2"
        }
]
```

### CODE REVIEW

File: bluemix/terminal/table_test.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import (
+	"bytes"
+	"github.com/stretchr/testify/assert"
+	"testing"
+
+	. "github.com/IBM-Cloud/ibm-cloud-cli-sdk/bluemix/terminal"
+)
+
+func TestPrintTableSimple(t *testing.T) {
```
Comment (jorge-ibm) [inline bluemix/terminal/table_test.go:11]: Thanks for adding these tests. Though I want to ask if you can maybe add some negative tests? What happens if the headers are empty strings ? Just want to make sure this new functionality is robust.

**Review by jorge-ibm — CHANGES_REQUESTED** [review]
Thanks for adding this functionality @allmightyspiff! Looks like there's some gosec failures that need to be addressed.

**Comment by allmightyspiff** [comment]
@jorge-ibm the gosec errors seems to be in unrelated files.  Although since both errors look like they are supposed to create a file from a variable, these might be false positives? 

I'll also add some more tests as well, so no worries there.

```
$  gosec -exclude-dir=plugin_examples -exclude-dir=vendor -quiet ./...
Results:


[ibm-cloud-cli-sdk\common\file_helpers\gzip.go:67] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
    66:
  > 67:                 f, err := os.Create(path)
    68:                 if err != nil {



[ibm-cloud-cli-sdk\common\file_helpers\file.go:55] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
    54:
  > 55:         destFile, err := os.Create(dest)
    56:         if err != nil {
```

**Comment by jorge-ibm** [comment]
thanks @allmightyspiff. I'll need to take a look at the gosec failures then. Also I just noticed, but can you please re-open this PR against the `dev` branch ?

**Comment by allmightyspiff** [comment]
Ok, I've rebased to `dev`, added more unit tests checking a few more "non-optimal" table setups and marked the 2 gosec violations as false positives.

my interpretation of the G304 error is that it only really applies if the go process has higher privileges than the user supplying the input, which I don't think is the case for the cli.

File: common/file_helpers/file.go [hunk 2]
```diff
@@ -52,6 +52,7 @@ func CopyFile(src string, dest string) (err error) {
 		return fmt.Errorf("%s is not a regular file.", src)
 	}
 
+	//#nosec G304 -- This is a false positive
```
Comment (jorge-ibm) [inline common/file_helpers/file.go:55]: I think this should be fine given we clean the path prior.

**Review by jorge-ibm — APPROVED** [review]

**Review by boyang9527 — APPROVED** [review]
